### PR TITLE
WASM native FileSavePicker, upload FileOpenPicker

### LINKED ISSUE #5232: [WASM] FileSavePicker
Part of #3215

## What would you like to be added:

`FileSavePicker` on WASM.

## Why is this needed:

💾

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #5339: [WASM] Upload-based FileOpenPicker
## What would you like to be added:

Support for upload-based `FileOpenPicker`

## Why is this needed:

Uploads!

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?

Part of #4028

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #5339, closes #5232

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported.

## What is the new behavior?

💥

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/AddIns/Uno.UI.Lottie/Uno.UI.Lottie.Wasm.csproj [hunk 1]
```diff
@@ -25,6 +25,11 @@
 		<UpToDateCheckInput Include="ts\**\*" />
 		<ConfigFiles Include="tsconfig.json" />
 	</ItemGroup>
+
+	<ItemGroup>
+		<None Remove="tsconfig.json" />
+		<Content Include="tsconfig.json" />
+	</ItemGroup>
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/AddIns/Uno.UI.Lottie/Uno.UI.Lottie.Wasm.csproj:32]: For now VS does not pick up on the `<ConfigFiles>` declaration, so it seems this is still needed, at least temporarily.

File: src/Uno.UI/WasmScripts/Uno.UI.js [hunk 2]
```diff
@@ -3200,6 +3238,40 @@ var Windows;
...
+                        options.types.push(pickerAcceptType);
+                    }
+                    try {
+                        const selectedFile = await showSaveFilePicker(options);
+                        const info = Uno.Storage.NativeStorageItem.getInfos(selectedFile)[0];
+                        const json = JSON.stringify(info);
+                        return json;
+                    }
+                    catch (e) {
+                        console.log("User did not make a selection or it file selected was" +
```
Comment (davidjohnoliver) [inline src/Uno.UI/WasmScripts/Uno.UI.js:3270]:
```suggestion
                        console.log("User did not make a selection or the file selected was" +
```

File: src/Uno.UI/ts/Windows/Storage/Pickers/FileSavePicker.ts [hunk 3]
```diff
@@ -1,6 +1,49 @@
...
+			}
+
+			try {
+				const selectedFile = await showSaveFilePicker(options);
+				const info = Uno.Storage.NativeStorageItem.getInfos(selectedFile)[0];
+				const json = JSON.stringify(info);
+				return json;
+			}
+			catch (e) {
+				console.log("User did not make a selection or it file selected was" +
```
Comment (davidjohnoliver) [inline src/Uno.UI/ts/Windows/Storage/Pickers/FileSavePicker.ts:41]:
```suggestion
				console.log("User did not make a selection or the file selected was" +
```

File: src/Uno.UWP/Storage/Pickers/FileSavePicker.wasm.cs [hunk 4]
```diff
@@ -4,26 +4,83 @@
...
+
+			var info = JsonHelper.Deserialize<NativeStorageItemInfo>(nativeStorageItemInfo);
+			return StorageFile.GetFromNativeInfo(info);
+		}
+
+		private NativeFilePickerAcceptType[] BuildFileTypesMap()
+		{
+			var acceptTypes = new List<NativeFilePickerAcceptType>();
+
+			var mimeTypeMap = new Dictionary<string, List<string>>();
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Storage/Pickers/FileSavePicker.wasm.cs:58]: > Remove the unused local variable 'mimeTypeMap'.

File: src/Uno.UWP/WinRTFeatureConfiguration.cs [hunk 5]
```diff
@@ -72,5 +72,19 @@ public static class FileTypes
...
 			public static Dictionary<string, string> FileTypeToMimeMapping { get; } = new Dictionary<string, string>();
 		}
+
+		public static class Storage
+		{
+			public static class Pickers
+			{
+				/// <summary>
+				/// Gets or sets a value indicating whether the file pickers based on
+				/// JS File System Access API are allowed. When set to false,
```
Comment (davidjohnoliver) [inline src/Uno.UWP/WinRTFeatureConfiguration.cs:82]:
```suggestion
				/// JS File System Access API are used. When set to false, or
				/// if the user's browser doesn't support the newer API,
```

**Review by davidjohnoliver — APPROVED** [review]

File: src/Uno.UI/ts/Windows/Storage/Pickers/FileOpenPicker.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+			return typeof showOpenFilePicker === "function";
+		}
+
+		public static async nativePickFilesAsync(
+			multiple: boolean,
+			showAllEntry: boolean,
+			fileTypesJson: string): Promise<string> {
+
+			if (!FileOpenPicker.isNativeSupported()) {
+				return JSON.stringify([]);
```
Comment (carldebilly) [inline src/Uno.UI/ts/Windows/Storage/Pickers/FileOpenPicker.ts:14]: `return "[]";` should be enough here ;-)

Comment (MartinZikmund) [reply src/Uno.UI/ts/Windows/Storage/Pickers/FileOpenPicker.ts:14]: Will fix as part of an upcoming PR :-)

**Pull request merged**